src/core/nginx.go

```diff
@@ -290,15 +294,27 @@ func (n *NginxBinaryType) Reload(processId, bin string) error {
 }
 
 // ValidateConfig tests the config with nginx -t -c configLocation.
-func (n *NginxBinaryType) ValidateConfig(processId, bin, configLocation string, config *proto.NginxConfig, configApply *sdk.ConfigApply) error {
-	log.Debugf("Validating config, %s for nginx process, %s", configLocation, processId)
-	response, err := runCmd(bin, "-t", "-c", configLocation)
+func (n *NginxBinaryType) ValidateConfig(processId, bin, configLocation string, errorLogPaths []string, config *proto.NginxConfig, configApply *sdk.ConfigApply) error {
+	log.Debugf("Validating config, %s for nginx process, %s, with custom error path %v", configLocation, processId, errorLogPaths)
+	var response *bytes.Buffer
+	var err error
+	cmdArgs := make([]string, 0)
+	cmdArgs = append(cmdArgs, "-t")
+	cmdArgs = append(cmdArgs, "-c")
+	cmdArgs = append(cmdArgs, configLocation)
+	if len(errorLogPaths) > 0 {
```

Comment:
no need for the if statement here

Reply:
I think we do, because at L311, we are doing  ```response, err = runCmd(bin, cmdArgs...)``` where `cmdArgs` is passed as an array.